src/googleError.ts

```diff
@@ -25,6 +25,32 @@ export class GoogleError extends Error {
   statusDetails?: string | protobuf.Message<{}>[];
   reason?: string;
   domain?: string;
+  errorInfoMetadata?: {string: string};
+
+  // Parse details field in google.rpc.status wire over gRPC medatadata.
+  // Promote google.rpc.ErrorInfo if exist.
+  static parseGRPCStatusDetails(err: GoogleError): GoogleError {
+    const decoder = new GoogleErrorDecoder();
+    try {
+      if (err.metadata && err.metadata.get('grpc-status-details-bin')) {
+        const statusDetailsObj: GRPCStatusDetailsObject =
+          decoder.decodeGRPCStatusDetails(
+            err.metadata.get('grpc-status-details-bin') as []
+          );
+        if (statusDetailsObj && statusDetailsObj.details.length > 0) {
```

Comment:
for extra safety: `statusDefailsObj && statusDetailsObj.details && statusDetailsObj.details.length > 0`, for cases where `details` is undefined

Reply:
Added 🙏